core/src/main/java/org/apache/carbondata/core/datamap/DataMapProvider.java

```diff
@@ -106,13 +122,206 @@ public void initData() { }
    * 1. after datamap creation and if `autoRefreshDataMap` is set to true
    * 2. user manually trigger REBUILD DATAMAP command
    */
-  public abstract void rebuild() throws IOException, NoSuchDataMapException;
+  public boolean rebuild() throws IOException, NoSuchDataMapException {
+    String newLoadName = "";
+    String segmentMap = "";
+    AbsoluteTableIdentifier dataMapTableAbsoluteTableIdentifier = AbsoluteTableIdentifier
+        .from(dataMapSchema.getRelationIdentifier().getTablePath(),
```

Comment:
Check for `getRelationIdentifier` and return false directly if it is null.

Reply:
okay. done